src/Tizen.Applications.Common/Tizen.Applications/AppControl.cs

```diff
@@ -640,6 +642,113 @@ public static void SendTerminateRequest(AppControl terminateRequest)
+        /// Also, implicit launch requests are NOT delivered to service applications since 2.4.
+        /// To launch a service application, an explicit launch request with the application ID given by property ApplicationId MUST be sent.
+        /// </remarks>
+        /// <param name="request">The AppControl.</param>
+        /// <param name="resultCallback">The callback function to be called when the result is delivered.</param>
+        /// <param name="replyCallback">The callback function to be called when the reply is delivered.</param>
+        /// <exception cref="ArgumentException">Thrown when failed because of the argument is invalid.</exception>
+        /// <exception cref="Exceptions.AppNotFoundException">Thrown when the application to run is not found.</exception>
+        /// <exception cref="Exceptions.LaunchFailedException">Thrown when the request failed to launch the application.</exception>
+        /// <exception cref="Exceptions.LaunchRejectedException">Thrown when the launch request is rejected.</exception>
+        /// <exception cref="Exceptions.OutOfMemoryException">Thrown when the memory is insufficient.</exception>
+        /// <exception cref="Exceptions.PermissionDeniedException">Thrown when the permission is denied.</exception>
+        /// <exception cref="TimeoutException">Thrown when failed because of timeout.</exception>
+        /// <privilege>http://tizen.org/privilege/appmanager.launch</privilege>
+        /// <since_tizen> 6 </since_tizen>
+        public static void SendLaunchRequestAsync(AppControl request, AppControlResultCallback resultCallback, AppControlReplyCallback replyCallback)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            if (resultCallback == null)
+            {
+                throw new ArgumentNullException(nameof(resultCallback));
+            }
+
+
+            int resultId = 0;
+            Interop.AppControl.ErrorCode err;
+
+            lock (_resultNativeCallbackMaps)
+            {
+                resultId = _resultNativeCallbackId++;
+                _resultNativeCallbackMaps[resultId] = (launchRequestHandle, result, userData) =>
+                {
+                    Log.Debug(LogTag, "Result callback is invoked");
+                    resultCallback(new AppControl(launchRequestHandle), (AppControlResult)result);
+                    lock (_resultNativeCallbackMaps)
+                    {
+                        _resultNativeCallbackMaps.Remove(resultId);
+                    }
+                };
+            }
+
+            if (replyCallback != null)
+            {
+                int replyId = 0;
+                lock (_replyNativeCallbackMaps)
+                {
+                    replyId = _replyNativeCallbackId++;
+                    _replyNativeCallbackMaps[replyId] = (launchRequestHandle, replyRequestHandle, result, userData) =>
```

Comment:
How about use a same Interop.AppControl.ReplyCallback and getting a user calllback by userData?
``` c#
 // a static constructor
 s_replyNativeCallback = (launchHandle, replyHandle, result, userData) => {
     int key = (int)userData;
    if (_replyCallbackMaps.Contains(key))
    {
           _resultCallbackMaps[key](new AppControl(launchRequestHandle), new AppControl(replyRequestHandle), (AppControlReplyResult)result);
           _resultCallbackMaps.Remove(key);
     }
 }

....

 err = Interop.AppControl.SendLaunchRequestAsync(request._handle, s_resultNativeCallback, s_replyNativeCallback , (IntPtr)key);
```

Reply:
I'll consider it.

Reply:
Done